### PR TITLE
Support datetime filters for datetime range lt[e]

### PR DESCRIPTION
**Pull request by lpouvreau**
Updated the DateTime Range for the `lt[e]` requests to forward the full provided datetime to the API instead of replacing its time with `23:59:59`

What we are aiming with this MR is to update the datetime ranges parameters so:
- We query the API with a date format (`YYYY-MM-DD`) when params are being provided as a date
- We query the API with a datetime format (`YYYY-MM-DDTHH:mm:ssZ`) when params are being provided as a datetime

1. We want to keep validating the params with their corresponding type validation
2. We need a field that will format/validate the proper way an input data that could be either a date or a datetime

The cleanest way to perform this is to use the undocumented [UnionType](https://github.com/schematics/schematics/blob/master/schematics/types/union.py#L33) from the `schematics` library.
The issue with that is that we would need to bump its version to `2.0.0+` but we are having important breaking changes when doing so.

As an in-between solution, I added a custom type class `DateOrDateTimeType` that inherits from schematics `DateTimeType` but will serialize/deserialize objects as a date or datetime depending on what is provided + skip TZ validation when handling dates

### CODE REVIEW

**Review by m-daru — APPROVED** [review]

**Review by mbonnefoy — APPROVED** [review]
lgtm

**Comment by lpouvreau** [comment]
Fixes issue #67

File: tests/endpoints/v1/test_events.py [hunk 1]
```diff
@@ -60,17 +60,54 @@ def test_search_params_underscores(self, client):
...
+            verify=True,
+        )
+
+    @with_mock_client()
+    def test_search_datetimerange_params(self, client):
+        client.events.search(
+            start__gte="2016-03-01",
+            start__lt="2016-04-01",
+            start__tz="Pacific/Auckland",
+            end__gte="2016-05-01 13:13:13",
```
Comment (mbonnefoy) [inline tests/endpoints/v1/test_events.py:85]: Any specific reason why you pass a non iso format in this test?

Comment (mbonnefoy) [reply tests/endpoints/v1/test_events.py:85]: Note: this applies to many places, including the use cases.

**Review by mbonnefoy — APPROVED** [review]
Looks good. I just suggest using iso format dt in our examples and tests.

File: tests/endpoints/v1/test_events.py (hunk 1 above)
Comment (lpouvreau) [reply tests/endpoints/v1/test_events.py:85]: No specific reason.

I used this format as it seemed to be the one used in the related issues's example.

I'll update it now 🙂

**Pull request merged**